src/stormpath.social-login.js

```diff
@@ -0,0 +1,143 @@
+ *
+ * Provides the {@link stormpath.socialLoginService.$socialLogin $socialLogin} service.
+ *
+ * Currently, this provider does not have any configuration methods.
+ */
+angular.module('stormpath.socialLogin', ['stormpath.CONFIG'])
+.config(['$injector', 'STORMPATH_CONFIG', function $socialLoginProvider($injector, STORMPATH_CONFIG) {
+  /**
+   * @ngdoc object
+   *
+   * @name stormpath.socialLoginService.$socialLogin
+   *
+   * @description
+   *
+   * The social login service provides methods for letting users logging in with Facebook, Google, etc.
+   */
+  var socialLoginFactory = ['$http', '$q', '$injector', function socialLoginFactory($http, $q, $injector) {
+    var providers;
+
+    function SocialLoginService() {
+      return this;
+    }
+
+    /**
+     * @ngdoc function
+     *
+     * @name  stormpath.socialLoginService.$socialLogin#getProviders
+     *
+     * @methodOf stormpath.socialLoginService.$socialLogin
+     *
+     * @returns {promise}
+     *
+     * A promise that is resolved with the list of all available OAuth providers.
+     *
+     * @description
+     *
+     * Returns a list of all OAuth providers, provided by the `/oauth/providers` endpoint.
+     */
+    SocialLoginService.prototype.getProviders = function getProviders() {
+      var deferred = $q.defer();
+
+      if (providers) {
+        deferred.resolve(providers);
+        return deferred.promise;
+      }
+
+      $http.get(STORMPATH_CONFIG.getUrl('OAUTH_PROVIDERS_ENDPOINT')).then(function(response) {
+        providers = response.data;
+
+        Object.keys(providers).forEach(function(providerName) {
+          var provider = providers[providerName];
+          var service;
+
+          try {
+            service = $injector.get('$' + providerName + 'Login');
+          } catch (err) {
+            return;
+          }
+
+          service.clientId(provider.clientId);
+          providers[providerName].service = service;
+        });
+
+        deferred.resolve(providers);
+      }).catch(deferred.reject);
+
+      return deferred.promise;
+    };
+
+    return new SocialLoginService();
+  }];
+
+  $injector.get('$provide').factory(STORMPATH_CONFIG.SOCIAL_LOGIN_SERVICE_NAME, socialLoginFactory);
+}])
+
+/**
+ * @ngdoc directive
+ *
+ * @name stormpath.spSocialLogin:spSocialLogin
+ *
+ * @description
+ *
+ * Add this directive to a button or link in order to authenticate using an OAuth provider. The value should be an OAuth provider id such as google, github, facebook or linkedin.
+ *
+ * @example
+ *
+ * <pre>
+ * <div class="container">
+ *   <button sp-social-login="facebook" sp-scope="public_profile,email">Login with Facebook</button>
+ * </div>
+ * </pre>
+ */
+.directive('spSocialLogin', ['$socialLogin', '$auth', function($socialLogin, $auth) {
+  return {
+    link: function(scope, element, attrs) {
+      var provider;
+
+      $socialLogin.getProviders().then(function(providers) {
+        provider = providers[attrs.spSocialLogin].service;
```

Comment:
This is actually not a provider, but a provider service... Right? Also, no checking if `attrs.spSocialLogin` actually exists in `providers`.